### ISSUE TITLE
[2017.2.1.4 Patch Release] Hololens. BoundingBoxGizmoExample. Any scene selections cause NullReferenceException

### ISSUE THREAD

**Issue by jchristof** [Microsoft/MixedRealityToolkit-Unity#2009]
## Overview
Tested in Hololens emulator
[2017.2.1.4 Patch Release] Hololens. BoundingBoxGizmoExample. Clicks on objects or empty scene space cause exceptions
## Expected Behavior
Selecting an object shows a bounding box with app bar
## Actual Behavior
eventData.selectedObject produces NullReferenceException
```
public void OnInputClicked(InputClickedEventData eventData)
        {
            if (Isinteractable(eventData.selectedObject))
            {
                InputClicked(eventData.selectedObject, eventData);
            }
        }
```
## Steps to reproduce
New project. Load BoundingBoxGizmoExample scene. Build for Hololens. Click anywhere in the scene and observe the output window.
## Unity Editor Version
Unity_2017_2.1p2
## Mixed Reality Toolkit Release Version
[2017.2.1.4 Patch Release]

**Comment by davidkline-ms** [comment]
@jchristof, What version of the HoloLens OS are you running?

Thx!

**Comment by jchristof** [comment]
@davidkline-ms Hi David:
Windows version	14393.1358.x86fre.rs1_release.170602-2252

**Comment by cre8ivepark** [comment]
@jchristof Does it only happen in the emulator?

**Comment by davidkline-ms** [comment]
Assigning to @cre8ivepark

**Comment by davidkline-ms** [comment]
@jchristof, is the code snippet you provided something you added to a script?
Thanks!

**Comment by jchristof** [comment]
Have not tried it on device.

**Comment by jchristof** [comment]
No not my script it came from a clean install of the toolkit and example
project package.

**Comment by davidkline-ms** [comment]
found it. thanks! i'll was not seeing an exception on my flighted HoloLens. I will continue to investigate on one with the version you mentioned.

**Comment by jchristof** [comment]
Thanks guys!

**Comment by davidkline-ms** [comment]
I have not been able to repro this using the emulator. I recall fixing this issue for the 2017.2.1.4 release...

@jchristof, if you re-download the 2017.2.1.4 package, or try the may18_dev branch and can still reproduce the issue, please reopen it.

Thanks!
David

**Comment by davidkline-ms** [comment]
I stand corrected.... I just saw the exception.

**Comment by davidkline-ms** [comment]
my exception is occurring in OnSourceLost, but may be related to this.

**Comment by davidkline-ms** [comment]
Exception thrown: 'System.NullReferenceException' in Assembly-CSharp.dll
'MixedRealityToolkit-Unity.exe' (CoreCLR: CoreCLR_UWP_Domain): Loaded 'C:\Data\Users\DefaultAccount\AppData\Local\DevelopmentFiles\MixedRealityToolkit-UnityVS.Release_x86.davidkl\System.Diagnostics.StackTrace.dll'. Cannot find or open the PDB file.
NullReferenceException: Object reference not set to an instance of an object.
   at HoloToolkit.Unity.UX.BoundingBoxGizmoHandle.OnSourceLost(SourceStateEventData eventData)
   at HoloToolkit.Unity.InputModule.InputManager.<>c.<.cctor>b__123_8(ISourceStateHandler handler, BaseEventData eventData)
   at UnityEngine.EventSystems.ExecuteEvents.Execute[T](GameObject target, BaseEventData eventData, EventFunction`1 functor) 
(Filename: <Unknown> Line: 0)

**Comment by davidkline-ms** [comment]
This was released in the 2017.4.0.0 release candidate
